gtest/src/wasm_executor.rs

```diff
@@ -0,0 +1,349 @@
+use crate::empty_ext::ExtImplementedStruct;
+use core_processor::common::JournalNote;
+use gear_backend_wasmtime::{env::StoreData, funcs::FuncsHandler};
+use gear_core::{
+    env::{Ext, ExtCarrier},
+    ids::ProgramId,
+    memory::{PageBuf, PageNumber, WasmPageNumber},
+    message::IncomingMessage,
+    program::Program,
+};
+use std::{collections::BTreeMap, convert::TryInto};
+use wasmtime::{
+    Config, Engine, Extern, Func, Instance, Memory as WasmtimeMemory, MemoryType, Module, Store,
+    Val,
+};
+
+#[allow(dead_code)]
+pub struct WasmExecutor {
+    instance: Instance,
+    store: Store<StoreData<ExtImplementedStruct>>,
+    memory: WasmtimeMemory,
+    program_id: ProgramId,
+}
+
+impl WasmExecutor {
+    pub(crate) fn new(
+        source: ProgramId,
+        program: Program,
+        memory_pages: &BTreeMap<PageNumber, Box<PageBuf>>,
+        message: Option<IncomingMessage>,
+    ) -> Self {
+        let ext = ExtImplementedStruct::new(source, program.id(), message);
+        let ext_carrier = ExtCarrier::new(ext);
+        let store_data = StoreData {
+            ext: ext_carrier.cloned(),
+            termination_reason: None,
+        };
+
+        let config = Config::new();
+        let engine = Engine::new(&config).expect("Failed to create engine");
+        let mut store = Store::<StoreData<ExtImplementedStruct>>::new(&engine, store_data);
+
+        let module = Module::new(&engine, program.code().code()).expect("Failed to create module");
+
+        let mut memory =
+            WasmtimeMemory::new(&mut store, MemoryType::new(program.static_pages().0, None))
+                .expect("Failed to create memory");
+
+        let mut funcs = BTreeMap::<&'static str, Func>::new();
+        funcs.insert("alloc", FuncsHandler::alloc(&mut store, memory));
```

Comment:
I want to move code that is building funcs tree from  ```core-backend::wasmtime::env::WasmtimeEnvironment``` to new struct and reuse it here
Would be great to hear somebody's opinion about this idea

Reply:
Yes, we can use a shared function that creates this `BTreeMap` of functions.